ledger/db/src/ledger_trait.rs

```diff
@@ -81,4 +81,13 @@ pub trait Ledger: Send {
 
     /// Get the tx out root membership element from the tx out Merkle Tree.
     fn get_root_tx_out_membership_element(&self) -> Result<TxOutMembershipElement, Error>;
+
+    /// Get the latest block header, if any. NotFound if the ledger-db is empty.
+    fn get_latest_block(&self) -> Result<Block, Error> {
+        let num_blocks = self.num_blocks()?;
+        if num_blocks == 0 {
+            return Err(Error::NotFound);
+        }
+        Ok(self.get_block(num_blocks - 1)?)
```

Comment:
Optional/Equivalent: Return the result of `get_block` directly.

```suggestion
        if num_blocks == 0 {
            return Err(Error::NotFound);
        }
        self.get_block(num_blocks - 1)
```
or even
```suggestion
        return if num_blocks == 0 {
            Err(Error::NotFound)
        } else {
            self.get_block(num_blocks - 1)
```

Reply:
yeah clippy said the same thing :)

```
error: question mark operator is useless here
  --> ledger/db/src/ledger_trait.rs:91:9
   |
91 |         Ok(self.get_block(num_blocks - 1)?)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `self.get_block(num_blocks - 1)`
   |
   = note: `-D clippy::needless-question-mark` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_question_mark
```